### DISCUSSION TITLE
Problem with excluding value from query parameters

### DISCUSSION THREAD

**Discussion by ghost** [livewire/livewire#7614] [7. Bugs (archived)]
### Livewire version

v3.3.3

### Laravel version

v10.38.2

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

I have created nullable int property `$bathrooms` that I want to use also as query param. I wanted to exclude it from query when property is null. So I added `#[Url(except: null)]`. The code works just fine, but the problem occurs when I open a page with an already entered query parameter in the url address. For example `http://laravel.test/properties?bathrooms=2`. Then when I press the button to reset the property, the null appears in the url and is not omitted `http://laravel.test/properties?bathrooms=null`. If I then refresh the page with such an address, the following error occurs: `Cannot assign string to property App\Livewire\Simple::$bathrooms of type ?int`

### Repository or Wirebox URL

https://wirebox.app/b/yrd22

### Additional information

The Wirebox example does not reproduce this problem correctly, because query params cannot be used there.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
Steps to reproduce:

**Example**
```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use Livewire\Attributes\Url;

class TestPage2 extends Component
{
    #[Url(except: null)]
    public ?int $bathrooms = null;

    public function render()
    {
        return <<<'HTML'
            <div>
                <input type="number" wire:model.live="bathrooms" />
                <button wire:click="set('bathrooms', null)">Reset Bathrooms</button>
            </div>
        HTML;

    }
}
```

When updating the "bathroom" or using "Reset Bathrooms", all works as expected.

If loading in via queryString - works as expected.

Issue occurs when:
1) Load in via queryString
2) Action "Reset Bathrooms"
3) Error Thrown
3a) Notice queryString sets to "null" as a string

**Reply by lrljoe** [reply]
Correction to above steps to reproduce:

Step 1) Navigate to Page (No QueryString Params)
Step 2) Use the text box to manipulate the "bathrooms" one or more times
Step 3) Use the "Reset Bathrooms" button to prove that the querystring disappears correctly
Step 4) Enter a new (non-null) value for bathrooms, witness that queryString updates
Step 5) Refresh the page (witnessing that the "bathrooms" populates correctly)
Step 6) Use the "Reset Bathrooms" button
Step 7) Witness that queryString sets "bathrooms" to "null", and error is thrown
